### ISSUE TITLE
onMoveEnd triggered on right click even that mouse is not moved at all

### ISSUE THREAD

**Issue by lancemao** [wbkd/react-flow#3256]
### Describe the Bug

onMoveEnd triggered on right click even that mouse is not moved at all. Occurs only for the first time

### Steps to Reproduce the Bug or Issue

1. add a onMoveEnd callback
2. add contextmenu event by using `addEventListener("contextmenu", onContextMenu)` on document
3. start ReactFlow app and right click anywhere with no mouse moving
4. onMoveEnd is called

### Expected behavior

onMoveEnd not triggered

### Platform

- OS: [Windows]
- Browser: [Chrome]
- Version: [114.0.5735.199]

**Comment by checkomkar** [comment]
Can you provide more information about the issue, I was not able to reproduce this.

**Comment by lancemao** [comment]
Hi, sorry for not providing enough info. Here is a complete


```javascript
import './App.css';
import ReactFlow, { Background, Controls, SelectionMode } from 'reactflow';
import 'reactflow/dist/style.css';

import React, {
  useEffect,
  useRef,
} from 'react';

const nodes = [
  {
    id: '1',
    data: { label: 'Hello' },
    position: { x: 0, y: 0 },
    type: 'input',
  },
  {
    id: '2',
    data: { label: 'World' },
    position: { x: 100, y: 100 },
  },
];

export default function App() {

  const reactFlowWrapper = useRef(null);

  useEffect(() => {
    reactFlowWrapper.current.addEventListener("contextmenu", onContextMenu);

    return () => {
      reactFlowWrapper.current.removeEventListener("contextmenu", onContextMenu);
    };
  });

  const onContextMenu = (e) => {
    e.preventDefault();
    console.log('onContextMenu')
  }

  const onMouseClick = (e) => {
    console.log('onMouseClick')
  }

  const onMoveEnd = (e) => {
    console.log('onMoveEnd')
  }

  return (
    <div className='app-root' ref={reactFlowWrapper}>
      <div style={{ height: '100%' }}>
        <ReactFlow nodes={nodes} onMoveEnd={onMoveEnd} panOnDrag={[2]}
          selectionOnDrag={true}
          onMove={onMouseClick}
          onSelectionStart={onMouseClick}
          selectionMode={SelectionMode.Partial}
        >
          <Background />
          <Controls />
        </ReactFlow>
      </div>
    </div>
  );
}
```

**Comment by checkomkar** [comment]
I tried to replicate the issue, it seems it's working as expected. Please find the below video attached. 

I used your code in React Flow Basic example.  

[https://www.loom.com/share/cb609d8a78294af8b6040892c0d7f83e](https://www.loom.com/share/cb609d8a78294af8b6040892c0d7f83e)

Here is the code:
```typescript

import { use } from 'chai';
import { MouseEvent, useEffect, useRef } from 'react';
import ReactFlow, {
  MiniMap,
  Background,
  BackgroundVariant,
  Controls,
  ReactFlowProvider,
  Node,
  Edge,
  useReactFlow,
  Panel
} from 'reactflow';

const onNodeDrag = (_: MouseEvent, node: Node) => console.log('drag', node);
const onNodeDragStop = (_: MouseEvent, node: Node) => console.log('drag stop', node);
const onNodeClick = (_: MouseEvent, node: Node) => console.log('click', node);



const initialNodes: Node[] = [
  {
    id: '1',
    type: 'input',
    data: { label: 'Node 1' },
    position: { x: 250, y: 5 },
    className: 'light',
  },
  {
    id: '2',
    data: { label: 'Node 2' },
    position: { x: 100, y: 100 },
    className: 'light',
  },
  {
    id: '3',
    data: { label: 'Node 3' },
    position: { x: 400, y: 100 },
    className: 'light',
  },
  {
    id: '4',
    data: { label: 'Node 4' },
    position: { x: 400, y: 200 },
    className: 'light',
  },
];

const initialEdges: Edge[] = [
  { id: 'e1-2', source: '1', target: '2', animated: true },
  { id: 'e1-3', source: '1', target: '3' },
];

const defaultEdgeOptions = {};

const BasicFlow = () => {
  const instance = useReactFlow();

  const updatePos = () => {
    instance.setNodes((nodes) =>
      nodes.map((node) => {
        node.position = {
          x: Math.random() * 400,
          y: Math.random() * 400,
        };

        return node;
      })
    );
  };

  const reactFlowWrapper = useRef<HTMLDivElement>(null);
  const onContextMenu = () => console.log('context');

  useEffect(()=>{
    if(reactFlowWrapper?.current){
      reactFlowWrapper?.current.addEventListener("contextmenu", onContextMenu);
    }
    
    return () => {
      if(reactFlowWrapper?.current){
        reactFlowWrapper?.current.removeEventListener("contextmenu", onContextMenu);
      }
      
    };
  })

  const logToObject = () => console.log(instance.toObject());
  const resetTransform = () => instance.setViewport({ x: 0, y: 0, zoom: 1 });

  const toggleClassnames = () => {
    instance.setNodes((nodes) =>
      nodes.map((node) => {
        node.className = node.className === 'light' ? 'dark' : 'light';

        return node;
      })
    );
  };

  return (
    <div ref={reactFlowWrapper} className='test-class' style={{height:'100%'}}>
      <ReactFlow
        defaultNodes={initialNodes}
        defaultEdges={initialEdges}
        onNodeClick={onNodeClick}
        onNodeDragStop={onNodeDragStop}
        onNodeDrag={onNodeDrag}
        className="react-flow-basic-example"
        minZoom={0.2}
        maxZoom={4}
        fitView
        defaultEdgeOptions={defaultEdgeOptions}
        selectNodesOnDrag={false}
        elevateNodesOnSelect={false}
        onMoveEnd={() =>{console.log("moveEnd")}}
      >
        <Background variant={BackgroundVariant.Dots} />
        <MiniMap />
        <Controls />

        <Panel position="top-right">
          <button onClick={resetTransform}>reset transform</button>
          <button onClick={updatePos}>change pos</button>
          <button onClick={toggleClassnames}>toggle classnames</button>
          <button onClick={logToObject}>toObject</button>
        </Panel>
      </ReactFlow>
    </div>
  );
};

export default function App() {
  return (
    <ReactFlowProvider>
      <BasicFlow />
    </ReactFlowProvider>
  );
}

```

**Comment by lancemao** [comment]
Normally it does work

However, with the following options, Left-Click will select the nodes while Right-Drag pans the canvas. 

```javascript
selectionOnDrag={true}
selectionMode={SelectionMode.Partial}
panOnDrag={[2]}
```

**Comment by checkomkar** [comment]
@lancemao I have raised the PR for the same. You can checkout in the branch and test it. Thanks.

**Comment by moklick** [comment]
fixed in v11.8.0
